ui/src/views/Tasks/TaskGroup/index.jsx

```diff
@@ -254,6 +260,67 @@ export default class TaskGroup extends Component {
     this.props.history.push(`/tasks/groups/${taskGroupId}`);
   };
 
+  subscribeToMore = () => {
+    const { taskGroupId } = this.props.match.params;
+
+    this.props.data.subscribeToMore({
+      document: taskGroupSubscription,
+      variables: {
+        taskGroupId,
+        subscriptions: [
+          'tasksDefined',
+          'tasksPending',
+          'tasksRunning',
+          'tasksCompleted',
+          'tasksFailed',
+          'tasksException',
+        ],
+      },
+      updateQuery(previousResult, { subscriptionData }) {
+        const { tasksSubscriptions } = subscriptionData.data;
+        const newTask = tasksSubscriptions.status.task;
+        const isFromSameTaskGroupId =
+          newTask.status.taskGroupId === taskGroupId;
```

Comment:
This should always be true, right?  I think that if it's not, that's a signal that something's pretty broken (the browser is going to be consuming 5-10 messages/second!).  My GraphQL is still pretty weak, but I think the way this is designed ends up [back here](https://github.com/taskcluster/taskcluster/blob/6704c2694a011b35de304abab1f1161e3ba64d2c/services/web-server/src/resolvers/Tasks.js#L137-L138) where a proper routingKeyPattern is defined to select only tasks in this task group.

So, perhaps omit this or turn it into an assertion?

Reply:
This is to make sure data is not from another task group which can happen when a message is in flight and a user searches for a different task group. I'll add a comment.

Reply:
Ah, neat.  Thanks.